ocs_ci/helpers/helpers.py

```diff
@@ -632,41 +632,17 @@ def default_storage_class(
     Returns:
         OCS: Existing StorageClass Instance
     """
-    external = config.DEPLOYMENT["external_mode"]
-    custom_storage_class = config.ENV_DATA.get("custom_default_storageclass_names")
-    if custom_storage_class:
-        from ocs_ci.ocs.resources.storage_cluster import (
-            get_storageclass_names_from_storagecluster_spec,
-        )
-
-        resources = get_storageclass_names_from_storagecluster_spec()
 
-    if interface_type == constants.CEPHBLOCKPOOL:
-        if custom_storage_class:
-            try:
-                resource_name = resources[constants.OCS_COMPONENTS_MAP["blockpools"]]
-            except KeyError:
-                logger.error(
-                    f"StorageCluster spec doesn't have the custom name for '{constants.CEPHBLOCKPOOL}' storageclass"
-                )
-        else:
-            if external:
-                resource_name = constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_RBD
-            else:
-                resource_name = constants.DEFAULT_STORAGECLASS_RBD
-    elif interface_type == constants.CEPHFILESYSTEM:
-        if custom_storage_class:
-            try:
-                resource_name = resources[constants.OCS_COMPONENTS_MAP["cephfs"]]
-            except KeyError:
-                logger.error(
-                    f"StorageCluster spec doesn't have the custom name for '{constants.CEPHFILESYSTEM}' storageclass"
-                )
-        else:
-            if external:
-                resource_name = constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_CEPHFS
-            else:
-                resource_name = constants.DEFAULT_STORAGECLASS_CEPHFS
+    if interface_type in [
+        constants.CEPHBLOCKPOOL,
+        constants.OCS_COMPONENTS_MAP["blockpools"],
+    ]:
+        resource_name = storageclass_name(constants.OCS_COMPONENTS_MAP["blockpools"])
+    elif interface_type in [
+        constants.CEPHFILESYSTEM,
+        constants.OCS_COMPONENTS_MAP["cephfs"],
+    ]:
+        resource_name = storageclass_name(constants.OCS_COMPONENTS_MAP["cephfs"])
```

Comment:
does it cover also external mode SCs?

Reply:
Yes , It will work for external mode too.